homeassistant/components/elmax/config_flow.py

```diff
@@ -60,20 +120,205 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+                data_schema=DIRECT_SETUP_SCHEMA,
+                errors={"base": "network_error"},
+            )
+        except ElmaxBadLoginError:
+            return self.async_show_form(
+                step_id="direct_setup",
+                data_schema=DIRECT_SETUP_SCHEMA,
+                errors={"base": "invalid_auth"},
+            )
+
+        # Retrieve the current panel status. If this succeeds, it means the
+        # setup did complete successfully.
+        panel_status: PanelStatus = await client.get_current_panel_status()
 
+        # Make sure this is the only Elmax integration for this specific panel id.
+        await self.async_set_unique_id(panel_status.panel_id)
+        self._abort_if_unique_id_configured()
+
+        return await self._check_unique_and_create_entry(
+            unique_id=panel_status.panel_id,
+            title=f"Elmax Direct {panel_status.panel_id}",
+            data={
+                CONF_ELMAX_MODE: self._selected_mode,
+                CONF_ELMAX_MODE_DIRECT_HOST: self._panel_direct_hostname,
+                CONF_ELMAX_MODE_DIRECT_PORT: self._panel_direct_port,
+                CONF_ELMAX_MODE_DIRECT_SSL: self._panel_direct_use_ssl,
+                CONF_ELMAX_MODE_DIRECT_FOLLOW_MDNS: self._panel_direct_follow_mdns,
+                CONF_ELMAX_PANEL_PIN: self._panel_pin,
+                CONF_ELMAX_PANEL_ID: panel_status.panel_id,
+                CONF_ELMAX_MODE_DIRECT_SSL_CERT: self._panel_direct_ssl_cert,
+            },
+        )
+
+    async def async_step_direct_setup(self, user_input: dict[str, Any]) -> FlowResult:
+        """Handle the direct setup step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="direct_setup",
+                data_schema=DIRECT_SETUP_SCHEMA,
+                errors=None,
+            )
+
+        self._panel_direct_hostname = user_input[CONF_ELMAX_MODE_DIRECT_HOST]
+        self._panel_direct_port = user_input[CONF_ELMAX_MODE_DIRECT_PORT]
+        self._panel_direct_use_ssl = user_input[CONF_ELMAX_MODE_DIRECT_SSL]
+        self._panel_pin = user_input[CONF_ELMAX_PANEL_PIN]
+        self._panel_direct_follow_mdns = user_input[CONF_ELMAX_MODE_DIRECT_FOLLOW_MDNS]
+
+        tmp_schema = vol.Schema(
+            {
+                vol.Required(
+                    CONF_ELMAX_MODE_DIRECT_HOST, default=self._panel_direct_hostname
+                ): str,
+                vol.Required(
+                    CONF_ELMAX_MODE_DIRECT_PORT, default=self._panel_direct_port
+                ): int,
+                vol.Required(
+                    CONF_ELMAX_MODE_DIRECT_SSL, default=self._panel_direct_use_ssl
+                ): bool,
+                vol.Required(
+                    CONF_ELMAX_MODE_DIRECT_FOLLOW_MDNS,
+                    default=self._panel_direct_follow_mdns,
+                ): bool,
+                vol.Required(CONF_ELMAX_PANEL_PIN, default=self._panel_pin): str,
+            }
+        )
```

Comment:
Can't we use a dict for this to store it in? You can use `self.add_suggested_values` to get rid of the defaults